Source/Initialization/InjectorMomentum.H

```diff
@@ -492,4 +488,15 @@ private:
     Object object;
 };
 
+// In order for InjectorMomentum to be trivially copyable, its destructor
+// must be trivial.  So we have to rely on a custom deleter for unique_ptr.
+struct InjectorMomentumDeleter {
+    void operator () (InjectorMomentum* p) const {
+        if (p) {
+            p->clear();
+            delete p;
```

Comment:
if you want to be super sure that `if (p)` is guarding, you can set `p = nullptr;` afterwards in the deleters.

(You can then even spare the `if (p)` because `p->clear();` will always work and `delete nullptr;` is safe.